src/cpp/session/modules/SessionHelp.cpp

```diff
@@ -867,6 +868,50 @@ void handleSessionRequest(const http::Request& request, http::Response* pRespons
    }
 }
 
+void handlePythonHelpRequest(const http::Request& request,
+                             http::Response* pResponse)
+{
+   // get URL (everything after 'python/' bit
+   std::string code = request.uri().substr(::strlen("/python/"));
+   if (code.empty())
+   {
+      pResponse->setError(http::status::BadRequest, "Malformed URL");
+      return;
+   }
+
+   // construct HTML help file from requested object
+   std::string path;
+   Error error = r::exec::RFunction(".rs.python.generateHtmlHelp")
+         .addParam(code)
+         .call(&path);
+   if (error)
+      LOG_ERROR(error);
+   
+   if (path.empty())
+   {
+      pResponse->setNotFoundError(request.uri());
+      return;
+   }
+   
+   FilePath filePath = module_context::tempDir().complete(path);
+   if (!filePath.exists())
+   {
+      pResponse->setNotFoundError(request.uri());
+      return;
+   }
+
+   // read file contents and serve
+   std::string contents;
+   error = core::readStringFromFile(filePath, &contents);
+   if (error)
+      LOG_ERROR(error);
+   
+   pResponse->setContentType("text/html");
+   pResponse->setNoCacheHeaders();
```

Comment:
Is there a reason we can't cache these? We usually allow help content to be cached. See here:

https://github.com/rstudio/rstudio/blob/5719361179d1020dc3157c4e24b21bcd17c483e6/src/cpp/session/modules/SessionHelpHome.cpp#L61.

Alternately, you might also be able to avoid having to read the file's contents yourself by using `setFile`.